In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import warnings
warnings.filterwarnings('ignore')
import statsmodels.api as sm
from statsmodels.formula.api import ols
from scipy.stats import ttest_ind


In [2]:
data = pd.read_excel('anova_lab_data.xlsx')

In [3]:
data

Power   Etching Rate
0   160 W          5.43
1   180 W          6.24
2   200 W          8.79
3   160 W          5.71
4   180 W          6.71
5   200 W          9.20
6   160 W          6.22
7   180 W          5.98
8   200 W          7.90
9   160 W          6.01
10  180 W          5.66
11  200 W          8.15
12  160 W          5.59
13  180 W          6.60
14  200 W          7.55

In [4]:
data.describe()

Etching Rate
count     15.000000
mean       6.782667
std        1.228643
min        5.430000
25%        5.845000
50%        6.240000
75%        7.725000
max        9.200000

From the information above we can deduce : 
n = 15

In [6]:
data.rename(columns={'Power ': 'power', 'Etching Rate': 'etching_rate'}, inplace=True)

In [7]:
data.groupby('power').agg(np.mean)

etching_rate
power              
160 W         5.792
180 W         6.238
200 W         8.318

From an initial point of view we can assume a machine with power 200W can etch faster than the others. 


Ho : the means are similar ; all powers have a similar etching rate (160w = 180w = 200w)

H1 : the means are different ; the power has a direct influence on the etching rate 


Hence the information collected is :

n = 15 -> sample

alpha = 5% -> we decide percentage

df of model = 3 - 1 = 2 -> nb categories -1

error terms = 15 - 3 = 12  -> sample - categories

Total DoF = 15-2 = 13 -> we are working with a 2tail model

In [8]:
model = ols('etching_rate ~ C(power)',data=data).fit()
sm.stats.anova_lm(model)

df     sum_sq   mean_sq          F    PR(>F)
C(power)   2.0  18.176653  9.088327  36.878955  0.000008
Residual  12.0   2.957240  0.246437        NaN       NaN

In [9]:
crit_value = stats.f.ppf(q = 1-.05, dfn= 2, dfd = 12) #check how the critical value reflects as well
crit_value

3.8852938346523933

In [10]:
power_a = data[data['power'] == '200 W']['etching_rate']

for power in data['power'].unique():
    power_b = data[data['power'] == power]['etching_rate']
    print(power, ttest_ind(power_a, power_b))

160 W Ttest_indResult(statistic=7.611403634613074, pvalue=6.237977344615716e-05)
180 W Ttest_indResult(statistic=5.827496614588661, pvalue=0.0003926796476049085)
200 W Ttest_indResult(statistic=0.0, pvalue=1.0)


P-value for 160W and 180W are lower than our confidence interval 0.05. Also, f-statistic is above 3.89 for both 160 and 180W power machines. Thus, we reject the null hypothesis. 